fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanTransactionProcessingServiceImpl.java

```diff
@@ -221,7 +222,11 @@ public LoanScheduleDTO getRecalculatedSchedule(final ScheduleGeneratorDTO genera
     public OutstandingAmountsDTO fetchPrepaymentDetail(final ScheduleGeneratorDTO scheduleGeneratorDTO, final LocalDate onDate, Loan loan) {
         OutstandingAmountsDTO outstandingAmounts;
 
-        if (loan.isInterestBearingAndInterestRecalculationEnabled() && !loan.isChargeOffOnDate(onDate)) {
+        Optional<LoanTransaction> activeReageTransactionOnDate = loanTransactionService.getActiveReageTransactionOnDate(loan, onDate);
+
+        if (loan.isInterestBearingAndInterestRecalculationEnabled() && !loan.isChargeOffOnDate(onDate)
+                && (activeReageTransactionOnDate.isEmpty() || LoanReAgeInterestHandlingType.DEFAULT
```

Comment:
its incorrect. What you need is if the loan is reaged AND EQUAL AMORTIZATION, then use the `getTotalOutstandingLoan()`